core/scratch_block_comment.js

```diff
@@ -434,3 +535,56 @@ Blockly.ScratchBlockComment.prototype.showContextMenu_ = function(e) {
   menuOptions.push(Blockly.ContextMenu.commentDeleteOption(this, Blockly.Msg.DELETE));
   Blockly.ContextMenu.show(e, menuOptions, this.block_.RTL);
 };
+
+/**
+ * Encode a comment subtree as XML with XY coordinates.
+ * @param {boolean=} opt_noId True if the encoder should skip the comment id.
+ * @return {!Element} Tree of XML elements.
+ * @package
+ */
+Blockly.ScratchBlockComment.prototype.toXmlWithXY = function() {
+  var element = goog.dom.createDom('comment');
+  element.setAttribute('id', this.id);
+  element.textContent = this.text_;
+  element.setAttribute('x', Math.round(this.x));
```

Comment:
`this.x` and `this.y` -> `this.x_` and `this.y_`

Reply:
Good catch!!